### PR TITLE
Fix for #71596

### LINKED ISSUE #71596: Traffic loss with externalTrafficPolicy:Local and proxy-mode=ipvs
**What happened**:

When a service with `externalTrafficPolicy:Local` is used with proxy-mode=ipvs only 1/nodes traffic gets through. The service have "type: LoadBalancer" and the loadBalancerIP is used for access.

If the service is accessed through the `NodePort` it works as expected.

The problem is that `kube-proxy` setup targets for **all** enpoints in ipvs for the loadBalancerIP while still disable the SNAT. So all traffic that does not happen to hit the own node is lost. Example;
```
> ipvsadm -L -n   # (narrowed)
TCP  192.168.1.4:32713 rr
  -> 11.0.4.3:8080                Masq    1      0          0         
TCP  10.0.0.0:8080 rr
  -> 11.0.1.2:8080                Masq    1      0          0         
  -> 11.0.2.2:8080                Masq    1      0          0         
  -> 11.0.3.2:8080                Masq    1      0          0         
  -> 11.0.4.3:8080                Masq    1      0          0         
```
Here the lbIP `10.0.0.0` gets all endpoints as target (wrong!) but the NotePort entry (`32713`) gets only the local endpoint as target (correct!).

The service manifest;
```
apiVersion: v1
kind: Service
metadata:
  name: mconnect-local
spec:
  selector:
    app: mconnect
  ports:
  - name: mconnect
    port: 5001
  - name: http
    port: 8080
  externalTrafficPolicy: Local
  type: LoadBalancer
```

[metallb](https://github.com/google/metallb) is used to obtain the lbIP.

**What you expected to happen**:

The targets for the `loadBalancerIP` should be the pods executing on the local node **only**, same as for NodePort and no traffic should be lost when using the `loadBalancerIP`.

**How to reproduce it (as minimally and precisely as possible)**:

* Use k8s with proxy-mode=ipvs
* Start a service with `externalTrafficPolicy:Local`
* Try to access the servcie via the loadBalancerIP
* Do `ipvsadm -L -n` on a node and compare entries for loadBalancerIP and NodePort

**Anything else we need to know?**:

Public clouds using an external LB that distributes to the NodePort will not see this bug.

I usually runs on `master` and the problem exist on `` I backed to 1.12 to see if it was a new problem.

**Environment**:
- Kubernetes version (use `kubectl version`):
```
Client Version: version.Info{Major:"1", Minor:"12", GitVersion:"v1.12.0", GitCommit:"0ed33881dc4355495f623c6f22e7dd0b7632b7c0", GitTreeState:"clean", BuildDate:"2018-09-27T17:05:32Z", GoVersion:"go1.10.4", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"12", GitVersion:"v1.12.0", GitCommit:"0ed33881dc4355495f623c6f22e7dd0b7632b7c0", GitTreeState:"clean", BuildDate:"2018-09-27T16:55:41Z", GoVersion:"go1.10.4", Compiler:"gc", Platform:"linux/amd64"}
```
- Cloud provider or hardware configuration:
  No cloud provider. [xcluster](https://github.com/Nordix/xcluster) on Ubuntu 18.04
- OS (e.g. from /etc/os-release):
  Xcluster
- Kernel (e.g. `uname -a`):
  `Linux vm-004 4.18.5 #1 SMP Fri Nov 30 08:47:18 CET 2018 x86_64 GNU/Linux`
- Install tools:
  -
- Others:
  Test using [mconnect](https://github.com/Nordix/mconnect)

/kind bug

### PR DESCRIPTION
**Pull request by lbernail**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it:**

To make `externalTrafficPolicy: Local` work with proxy-mode=ipvs an loadBalancerIP.
The former test took into account sessionAffinity to only route to local endpoints but we should only consider local endpoints regardless of sessions affinity.

**Which issue(s) this PR fixes**
Fixes #71596

**Note**
Small update on https://github.com/kubernetes/kubernetes/pull/71610 by @uablrek 

Does this PR introduce a user-facing change?:
```release-note
IPVS: "ExternalTrafficPolicy: Local" now works with LoadBalancer services using loadBalancerIP
```

/assign @m1093782566

### CODE REVIEW

File: pkg/proxy/ipvs/proxier.go [hunk 1]
```diff
@@ -1008,11 +1008,9 @@ func (proxier *Proxier) syncProxyRules() {
 					serv.Timeout = uint32(svcInfo.StickyMaxAgeSeconds)
 				}
 				if err := proxier.syncService(svcNameString, serv, true); err == nil {
-					// check if service need skip endpoints that not in same host as kube-proxy
-					onlyLocal := svcInfo.SessionAffinityType == v1.ServiceAffinityClientIP && svcInfo.OnlyNodeLocalEndpoints
```
Comment (yastij) [inline pkg/proxy/ipvs/proxier.go]: This change seems reasonable to me, as you don't need to have serviceAffinity and ExternalTrafficPolicy to only sync local endpoints. From what I see #64146 added this to fix flakiness of an e2e. Let's give e2e some runs and see how it goes.

**Comment by nikhita** [comment]
/assign @m1093782566
for approval

File: pkg/proxy/ipvs/proxier.go (hunk 1 above)
Comment (m1093782566) [reply pkg/proxy/ipvs/proxier.go]: @yastij I would like to know the e2e test results. Have you ever run the tests?

Comment (lbernail) [reply pkg/proxy/ipvs/proxier.go]: @m1093782566 isn't /test all enough?

Comment (m1093782566) [reply pkg/proxy/ipvs/proxier.go]: I assume not, I usually use `kubetest --provider=gce --test` to run e2e tests on GCE - this PR is GCE/AWS LB related. Please reference: https://github.com/kubernetes/community/blob/master/contributors/devel/e2e-tests.md

Comment (yastij) [reply pkg/proxy/ipvs/proxier.go]: @lbernail - nope, not all e2e are played. I did a run for the relevant e2e. will add them later today.

Comment (lbernail) [reply pkg/proxy/ipvs/proxier.go]: Thanks! I assumed e2e tests were run as part of the standard tests

**Comment by MrHohn** [comment]
> ExternalTrafficPolicy: Local now works with LoadBalancer services using loadBalancerIP

Probably make the release note clear this is fixing the ipvs mode?

**Comment by MrHohn** [comment]
@lbernail Looks great, thanks!

**Comment by yastij** [comment]
```
[sig-network] ESIPP [Slow] [DisabledForLargeClusters] should work for type=LoadBalancer
[sig-network] ESIPP [Slow] [DisabledForLargeClusters] should work for type=NodePort
[sig-network] Services should have session affinity work for LoadBalancer service with ESIPP on [Slow] [DisabledForLargeClusters]
[sig-network] ESIPP [Slow] [DisabledForLargeClusters] should handle updates to ExternalTrafficPolicy field
[sig-network] Services  should be able to switch session affinity for LoadBalancer service with ESIPP off [Slow] [DisabledForLargeClusters]
[sig-network] Services  should be able to switch session affinity for LoadBalancer service with ESIPP on [Slow] [DisabledForLargeClusters]
[sig-network] ESIPP [Slow] [DisabledForLargeClusters] should work from pods
[sig-network] Services should have session affinity work for LoadBalancer service with ESIPP off [Slow] [DisabledForLargeClusters]
[sig-network] ESIPP [Slow] [DisabledForLargeClusters]

Ran 4 of 2119 Specs in 327.187 seconds
SUCCESS! -- 4 Passed | 0 Failed | 0 Pending | 2115 Skipped PASS
```
Ran this for a while now.

/lgtm

**Comment by yastij** [comment]
cc @m1093782566

**Comment by m1093782566** [comment]
Okay, great!

/approve

Let's keep watching the test grid after this PR goes in.

**Pull request merged**